packages/txservice/src/transaction.ts

```diff
@@ -158,35 +114,39 @@ export class Transaction {
     // Check to make sure that, if this is a replacement tx, the replacement gas is higher.
     if (this.responses.length > 0) {
       // NOTE: There *should* always be a gasPrice in every response, but it is
-      // defined as optional. Handle that case :(
+      // defined as optional. Handle that case?
       // If there isn't a lastPrice, we're going to skip this validation step.
       const lastPrice = this.responses[this.responses.length - 1].gasPrice;
       if (lastPrice && this.gasPrice.get().lte(lastPrice)) {
+        // NOTE: We do not set this._error here, as the transaction hasn't failed - just the txservice.
         throw new TransactionServiceFailure("Gas price was not incremented from last transaction.", { method });
       }
     }
 
     // Check to make sure we haven't already mined this transaction.
-    if (this.didFinish()) {
+    if (this.didFinish) {
+      // NOTE: We do not set this._error here, as the transaction hasn't failed - just the txservice.
       throw new TransactionServiceFailure("Submit was called but transaction is already completed.", { method });
     }
 
     // Increment transaction attempts made.
     this._attempt++;
+    this.timestamp = Date.now();
 
     // Send the tx.
     let result = await this.provider.sendTransaction(this.data);
 
     // If the error is a NonceExpired error and we haven't submitted yet, we want to keep
     // trying to send here. Reason being that it may take a few tries to get the correct
     // transaction count back from the provider.
```

Comment:
Couldn't this also mean you're trying to submit a tx at some nonce `n` that has already been confirmed at least once or has a gas bump in the mempool? This seems like it would be weird on restarts if the signer is frozen

Reply:
Yes, this model is entirely incompatible with the new model I have of assigning the nonce once we create the tx

Reply:
great spot!